Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -127,7 +127,8 @@ void RenderGrid::styleDidChange(StyleDifference diff, const RenderStyle* oldStyl
 bool RenderGrid::subgridDidChange(const RenderStyle& oldStyle) const
 {
     return oldStyle.gridSubgridRows() != style().gridSubgridRows()
-        || oldStyle.gridSubgridColumns() != style().gridSubgridColumns();
+        || oldStyle.gridSubgridColumns() != style().gridSubgridColumns()
+        || oldStyle.usedContain().contains(Containment::Layout) !=style().usedContain().contains(Containment::Layout);
```

Comment:
Is this change really necessary? I'd think rareDataChangeRequiresLayout should do the trick. (if not, we still need to find a new place for this line as it is not strictly a "sub grid change")

Reply:
I did try without the above change, but it did not work. Both changes were needed to prevent the assert

Reply:
ok, thanks for confirming. let me take a quick look.

Reply:
I believe when a subgrid ends up establishing an independent formatting context I *think*, at a minimum, we need to `setNeedsItemPlacement(true)` on it so that we end up recomputing the row and columns correctly. If we don't want this logic here or in `styleDidChange` maybe the parent grid could be responsible for checking this on its subgrids? @alanbaradlay

Reply:
I am just surprised that self_is_dirty itself does not invalidate item placement as it looks like anything that changes the layout boundary is going to trigger this RELEASE_ASSERT.
e.g. `document.getElementById("outer-subgrid").style.position = "absolute"; ` certainly does...so we should probably come up with a more robust solution here.